src/Internal/TunnelSocket.php

```diff
@@ -60,14 +61,20 @@ public function end(): void
         $this->localSocket->end();
     }
 
+    /** @api */
     public function reference(): void
     {
+        \assert($this->localSocket instanceof ResourceStream);
+        \assert($this->remoteSocket instanceof ResourceStream);
         $this->localSocket->reference();
         $this->remoteSocket->reference();
     }
 
+    /** @api */
```

Comment:
Why these annotations?

Reply:
It's to mark this unused function as public API so psalm doesn't complain about unused code